dynamic_programming/levenshtein_distance.nim

```diff
@@ -0,0 +1,68 @@
+## Levenshtein distance
+
+import tables
+
+
+func toKey(indA, indB: int): (Natural, Natural) =
+  return (indA.Natural, indB.Natural)
+
+
+func initSubsolutions(lenA, lenB: Natural): Table[(Natural, Natural), Natural] =
+  for indA in 0.Natural..lenA:
+    result[toKey(indA, 0)] = indA
+
+  for indB in 1.Natural..lenB:
+    result[toKey(0, indB)] = indB
+
+
+func computeLevenshteinDistanceMatrix(a, b: string): Table[(Natural, Natural),
+                                                           Natural] =
+  result = initSubsolutions(a.len, b.len)
+
+  for indA in 0..<a.len:
+    for indB in 0..<b.len:
```

Comment:
Partially done in 927d0c18ec0e7d835b98bc3f5fb62740a587f112. I am against putting this information at the top of the file, because this refers to the function and not to the file.